src/shared/hooks/population.ts

```diff
@@ -0,0 +1,75 @@
+import { RefObject, useCallback, useEffect, useState } from "react"
+import {
+  selectDisplayedPopulationById,
+  selectMaxDisplayedPopulation,
+  selectPopulationById,
+  selectSortedDisplayedPopulation,
+  setRealmDisplayedPopulation,
+  useDappDispatch,
+  useDappSelector,
+} from "redux-state"
+import { SECOND } from "shared/constants"
+import { calculatePopulationIconsPositions, randomInteger } from "shared/utils"
+import { useInterval } from "./helpers"
+
+export function usePopulationBubble(realmId: string): {
+  showBubble: boolean
+  setShowBubble: (newValue: boolean) => void
+} {
+  const population = useDappSelector((state) =>
+    selectPopulationById(state, realmId)
+  )
+  const displayedPopulation = useDappSelector((state) =>
+    selectDisplayedPopulationById(state, realmId)
+  )
+  const dispatch = useDappDispatch()
+
+  const [showBubble, setShowBubble] = useState(false)
+  // Generate random intervals for realms
+  const [delay] = useState(randomInteger(5, 15) * SECOND)
+
+  const populationCallback = useCallback(async () => {
+    if (population < displayedPopulation) {
+      await dispatch(
+        setRealmDisplayedPopulation({
+          id: realmId,
+          population,
+        })
+      )
+    } else if (population > displayedPopulation) {
+      setShowBubble(true)
```

Comment:
1. Shouldn't we dispatch new displayed population first and then show the bubble?
2. Don't we need to set the `showBubble` back to false at some point? I see that `Bubble` component is making if `false` when animation is finished so basically we will not show every population update because we will skip these happening during the animation, right?
3. Looking at code below:
```
      setRealmDisplayedPopulation({
          id: realmId,
          population: displayedPopulation + 1,
        })
```
That means the displayed population may start lagging a bit? I think we are not creating a queue for more population updates.

Reply:
1. Yes
2. From what I am seeing after playing the animation every bubble is appearing on the map, I didn't encounter a situation where population is incremented and bubble is not shown for this event